### PR TITLE
[scheduler] replaced soft polling with events driven scheme

### PR DESCRIPTION
**Pull request by dmitryermilov**
Signed-off-by: Dmitry Ermilov <dmitry.ermilov@intel.com>

### CODE REVIEW

**Comment by dmitryermilov** [comment]
-1 from myself. Minor change is needed.

File: _studio/mfx_lib/scheduler/src/mfx_scheduler_core_task_management.cpp [hunk 1]
```diff
@@ -296,7 +296,7 @@ bool mfxSchedulerCore::IsReadyToRun(MFX_SCHEDULER_TASK *pTask)
         // let the thread inspect other tasks.
         if (pTask->param.occupancy) {
             return false;
-        } else {
+        } else if (m_timeWaitPeriod) {
```
Comment (dmitryermilov) [inline _studio/mfx_lib/scheduler/src/mfx_scheduler_core_task_management.cpp]: I'm anticipating questions about this if() . The intention is to disable code block below and at the same time to simplify merge with Windows in the future.

Comment (dvrogozh) [reply _studio/mfx_lib/scheduler/src/mfx_scheduler_core_task_management.cpp]: Without this you will keep trying to execute the task which returned MFX_TASK_BUSY again and again. I.e. there is a risk of polling. We need to evaluate whether such polling can or can not happen.

File: _studio/mfx_lib/encode_hw/mpeg2/src/mfx_mpeg2_encode_full_hw.cpp [hunk 2]
```diff
@@ -498,13 +485,13 @@ mfxStatus FullEncode::QueryFrame(sExtTask2 *pExtTask)
     if(!is_initialized())
         return MFX_ERR_NOT_INITIALIZED;
 
-    if (pExtTask->m_nInternalTask == 0)
+    pIntTask = GetIntTask(pExtTask->m_nInternalTask);
+    MFX_CHECK_NULL_PTR1(pIntTask);
+
+    if (pIntTask->m_pBitstream == 0)
```
Comment (mgonchar) [inline _studio/mfx_lib/encode_hw/mpeg2/src/mfx_mpeg2_encode_full_hw.cpp]: nullptr

**Comment by dmitryermilov** [comment]
@ogolikov , please look at mpeg2e changes

**Pull request merged**